backend/data_adapter.go

```diff
@@ -2,23 +2,66 @@ package backend
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
 
+	"github.com/google/uuid"
 	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
 )
 
 // dataSDKAdapter adapter between low level plugin protocol and SDK interfaces.
 type dataSDKAdapter struct {
-	queryDataHandler QueryDataHandler
+	queryDataHandler  QueryDataHandler
+	conversionHandler ConversionHandler // Optional
 }
 
-func newDataSDKAdapter(handler QueryDataHandler) *dataSDKAdapter {
+func newDataSDKAdapter(handler QueryDataHandler, conversionHandler ConversionHandler) *dataSDKAdapter {
 	return &dataSDKAdapter{
-		queryDataHandler: handler,
+		queryDataHandler:  handler,
+		conversionHandler: conversionHandler,
 	}
 }
 
+func (a *dataSDKAdapter) ConvertQueryData(ctx context.Context, req *QueryDataRequest) (*QueryDataRequest, error) {
+	convertRequest := &ConversionRequest{
+		PluginContext: req.PluginContext,
+		// TODO: What should we use for UID and TargetVersion?
+		UID: uuid.New().String(),
```

Comment:
```suggestion
```
UID is not necessary for this -- that is the request UID used for tracking/tracing